sh/tests/better-cat_test.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+# Unofficial Bash Strict Mode
+set -euo pipefail
+IFS='
+'
+script_dir=$(cd -P "$(dirname "$BASH_SOURCE")" &>/dev/null && pwd)
+
+rm -rf better-cat-files
+mkdir better-cat-files
+cd better-cat-files
+
+for i in {1..3}; do
+    echo "This is line $i of file2" >> "file2.txt"
+done
+
+for i in {1..10}; do
+    echo "This is line $i of file3" >> "file3.txt"
+    echo "This is line $i of commented" >> "commented.txt"
+done
+
+echo "This is the content of file1" > "file1.txt"
+echo "# commented" > "commented.txt"
+
+challenge() {
+    submitted=$(bash "$script_dir"/student/better-cat.sh "$@")
+    expected=$(bash "$script_dir"/solutions/better-cat.sh "$@")
+    diff <(echo "$submitted") <(echo "$expected") 
+}
+
+challenge "file1.txt"
```

Comment:
What if you pass a file that does not exist?

Reply:
I didn't add it since the way cat behaves when a file does not exist is similar to the the default behaviour and the exercise is complex enough as it is.

I thought about adding a line clarifying that it won't be tested for inexistent files, but since in the description of the exercise that I made is not asking to handle this situation specifically I thought it would be redundant, specially considering this an optional exercise(not one of the very verbose almost step-by-step hints ones).

However if you strongly disagree let me know and I will make the changes :slightly_smiling_face:

Reply:
I don't have a strong opinion on this: I found that it is always easier to state what should happen when someone try to use the script in a "wrong way", but at the same time we can keep it simple and add a line if someone is going to complain in the future :)

Reply:
At this point, since the exercise description doesn't require handling inexistent files, I agree with you and I think it's reasonable to leave it as is. However, if any users have questions or concerns in the future, we can always revisit this topic and add a line to clarify the behaviour when a file doesn't exist. Thank you for your input and collaboration!